### ISSUE TITLE
Remote chat control for trusted players

### ISSUE THREAD

**Issue by strutzt** [cabaletta/baritone#1832]
## Describe your suggestion
On multiplayer servers, allow trusted players to send commands to Baritone instances via global chat or /msg.
Send output back to those trusted users.

All prefixed chat messages coming from non-trusted players should be handled like a normal chat message.

**Examples:**
`/msg BaritoneBot3 #mine diamond_ore` (applies to Baritone instance on account BaritoneBot3)
`/say #pause` (applies to all bots)

## Settings
Allow the user to configure trusted players for each server, for example in a local json file.
This is especially important on offline-mode servers, where multiple players may share the same username on different servers: A trusted username on one server might not correspond to the same person on another server.

## Context
It looks like #244 won't be merged anytime soon. This feature is much easier to implement and would still allow users to easily control multiple bots without switching clients. Bots could be running on different machines, even without a GUI and still be controllable.

This feature also allows for scripted automation: It's not hard to write a proxy or chatbot to inject chat messages depending on previous messages or a current state.

## Final checklist
- [x] I know how to properly use check boxes 
- [x] I have not used any OwO's or UwU's in this issue.

**Comment by dominikaaaa** [comment]
You can already do this

**Comment by strutzt** [comment]
How? I tried sending commands with and without prefix, that didn't do anything. Is there something I overlooked? I didn't find any settings or code related to something like this.

**Comment by dominikaaaa** [comment]
`#goto whatever`

**Comment by strutzt** [comment]
Are we talking about the same thing here? What I want to do is allow another player on the same server to send commands to my Baritone instance, not just control the client I'm currently on.
Let's say I have two accounts on the server, Notch and BaritoneBot, and I want to control BaritoneBot by sending commands via chat from Notch.
If I run something like `/msg BaritoneBot #goto lava` on the client logged in as Notch, the Baritone client logged in as BaritoneBot won't do anything. Just running `#goto lava` on Notch only controls the instance I'm running that on, or if Baritone isn't installed there just send that to global chat, which also does nothing on the BaritoneBot player.

**Comment by dominikaaaa** [comment]
Oh i'm pretty dumb, I completely misread the premise of this issue. Sorry about that.

**Comment by dominikaaaa** [comment]
>  or if Baritone isn't installed there just send that to global chat, which also does nothing on the BaritoneBot player.
How would it send it to global chat if baritone isn't installed tho

**Comment by strutzt** [comment]
Well, if Notch is on a vanilla client and types that in it will just send it to chat, because nothing's there to cancel the chat message

**Comment by dominikaaaa** [comment]
> Well, if Notch is on a vanilla client and types that in it will just send it to chat, because nothing's there to cancel the chat message

I'm not sure I follow. How would that work?

**Comment by strutzt** [comment]
If I'm not wrong, what Baritone is doing with those commands is listening for the user to send a chat message, checking if the message looks like a command and if that's the case cancel the chat message and instead execute it as a Baritone command. All other messages are ignored by Baritone and go straight to global chat like on a unmodified client, and if Baritone isn't there to do the filtering, messages with a command prefix will go to global chat too. So if you type `#goto lava` in chat, that's a command for baritone, but it's just a normal chat message for a vanilla client.

**Comment by dominikaaaa** [comment]
I don't see why you had to specify that then, it's literally "if baritone is installed the command won't do anything"

**Comment by yodaluca** [comment]
I think this would be a REALLY cool feature because then if you have lots of AlTS on laptops around your house or what not you don’t have to go around and type in the commands on those you can just remotely send it to them from your main computer.

**Comment by yodaluca** [comment]
And also umm so I’m new to git hub so I may not know if this is right but if this becomes a feature Dominika please add it to Impact client I noticed that your in the baritone one and Impact so idk if you are a developer for both or what but please add this to Impact.

**Comment by strutzt** [comment]
> I don't see why you had to specify that then, it's literally "if baritone is installed the command won't do anything"

I just wanted to be as specific as possible. At the time you had told me that it was already possible, so if what I described wouldn‘t have been the expected behavior, for example if Baritone was already handling global chat messages as commands, that would have been important to know. But I guess that‘s clear now.

**Comment by dominikaaaa** [comment]
I just added this into the latest KAMI Blue beta, if you don't want to wait for this to be added to mainstream Baritone

**Comment by strutzt** [comment]
That's a good start, thank you!

**Comment by dominikaaaa** [comment]
![ezgif com-optimize](https://user-images.githubusercontent.com/17222512/86054566-60942080-ba28-11ea-8764-855dca67fca1.gif)
it is, and that's about the extent that i will go because i really cba to reimplement this into baritone, even though it's like 1 mixin, it's all written in Kotlin and baritone is in Java

**Comment by RealIndrit** [comment]
KoTliN, bruh XD

**Comment by CrimsonXTaco** [comment]
Do both clients need KAMI Blue running? or just the receiving client? 

Apologize for necroposting

**Comment by l1ving** [comment]
> or just the receiving client?

Only receiving. Ask for help on the KAMI Blue Discord if you need more help with it

**Comment by dolphin-cat** [comment]
Cool idea, I like this!!!
You can run 21 minecrafts, 20 with baritone on a powerful computer and 1 without on your laptop, and you will be able to control 20 bots from anywhere you are if you have internet and play on the same server as the bots are, you can take this a step further with vnc, ssh and automation and make the 20 bots join a different server and do /login from anywhere you are if you have internet

**Comment by NullyIsHere** [comment]
@5HT2 Kami is now archived, is there any way to use it? I want to use it in new versions.

**Comment by Justinprow** [comment]
Yeah. I'd love for this to be added. I could really use another Minecraft instance with baritone installed, and use /msg or global chat, maybe with permissions, to control the bot account.
Msg by either server side or multiplayer mods.
This all can be added in settings, like '#allowremotecontrol true' or maybe addon mods? Whatever is the easiest or most efficient.
